src/app/_services/depublication.service.ts

```diff
@@ -0,0 +1,106 @@
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { apiSettings } from '../../environments/apisettings';
+import { MoreResults, RecordPublicationInfo, Results, SortParameter } from '../_models';
+
+import { collectResultsUptoPage } from './service-utils';
+import { ErrorService } from './error.service';
+
+@Injectable({ providedIn: 'root' })
+export class DepublicationService {
+  constructor(private readonly http: HttpClient, private readonly errors: ErrorService) {}
+
+  /** handleUploadEvents
+  /*  record upload events
+  /*  @param {HttpEvent} event - event from the server
+  */
+  handleUploadEvents(event: HttpEvent<Object>): boolean {
+    if (event.type === HttpEventType.Response) {
+      return true;
+    } else {
+      if (event.type === HttpEventType.UploadProgress) {
+        console.log(`File upload: ${event.loaded} / ${event.total}`);
+      }
+      return false;
+    }
+  }
+
+  /** setPublicationFile
+  /*  post publication information
+  /*  @param {string} datasetId - the dataset id
+  /*  @param {File} file - file of record urls
+  */
+  setPublicationFile(datasetId: string, file: File): Observable<boolean> {
+    const url = `${apiSettings.apiHostCore}/records/${datasetId}`;
+    const formData = new FormData();
+
+    formData.append('depublicationFile', file);
+
+    return this.http
+      .post(url, formData, {
+        headers: {
+          'Content-Type': file.type
+        },
+        observe: 'events',
+        params: {
+          clientFilename: file.name,
+          mimeType: file.type
+        },
+        reportProgress: true
+      })
+      .pipe(map(this.handleUploadEvents))
+      .pipe(this.errors.handleRetry());
+  }
+
+  /** setPublicationInfo
+  /*  post publication information
+  /*  @param {string} datasetId - the dataset id
+  /*  @param {string} toDepublish - depublication record urls
+  */
+  setPublicationInfo(datasetId: string, toDepublish: string): Observable<boolean> {
+    const url = `${apiSettings.apiHostCore}/records/${datasetId}`;
+    return this.http.post<boolean>(url, { toDepublish }).pipe(this.errors.handleRetry());
+  }
+
+  /** parseSortParameter
+  /*  return string representation of a SortParameter or an empty string
+  /*  @param {SortParameter} p - optional
+  */
+  parseSortParameter(p?: SortParameter): string {
+    return p ? `&sortField=${p.field}&sortDirection=${p.direction.toLowerCase()}` : '';
+  }
+
+  /** getPublicationInfo
+  /*  retrieve publication information
+  /*  @param {string} datasetId - the dataset id
+  /*  @param {number} page - the pagination target
+  /*  @param {SortParameter} sort - optional - sort parameter
+  */
+  getPublicationInfo(
+    datasetId: string,
+    page: number,
+    sort?: SortParameter
+  ): Observable<Results<RecordPublicationInfo>> {
+    const sortParam = this.parseSortParameter(sort);
+    const url = `${apiSettings.apiHostCore}/records/${datasetId}?nextPage=${page}${sortParam}`;
```

Comment:
1. Why call it `nextPage` and not `page`?
2. What should we do with the `UNSET` option that `SortDirection` can have? I think you (the UI) should tell me what to sort on, not what not to sort on.

Reply:
Oh, and:
3. Can we name the three sortable fields as follows:
- `recordId` (rather than `recordUrl` - it's not necessarily a URL)
- `depublicationStatus` (rather than `publicationStatus` - depublication is what we're after)
- `depublicationDate` (that one is fine as it is)

Reply:
>     1. Why call it `nextPage` and not `page`?
> 
>     2. What should we do with the `UNSET` option that `SortDirection` can have? I think you (the UI) should tell me what to sort on, not what not to sort on.

The value UNSET never gets sent to the server.  It's included as states/classes are derived from the ENUM on individual headers.  When the header-group handler sees UNSET it nullifies the param information sent to the server.